crates/volta-core/src/distro/package.rs

```diff
@@ -188,6 +189,15 @@ impl Distro for PackageDistro {
 
         let archive = self.load_or_fetch_archive()?;
 
+        let tmp_root = path::tmp_dir()?;
+        let temp = tempdir_in(&tmp_root)
+            .with_context(|_| ErrorDetails::CreateTempDirError { in_dir: tmp_root })?;
+        debug!(
```

Comment:
Just a suggestion: it feels like details of logging logic distract from the main line of logic in these functions, and it might be nice to extract them out into a helper method so you can inject them as a single line (e.g., `self.log_unpacking()` or `self.log(Verbose::Unpacking)` or something).

Reply:
Hmm, that's a bit of a balancing act. I think there are benefits to having the logging right in with the other code, for ease of searching when something goes wrong. But having them be single lines is nice, so I went with pulling out the ones in this file that fell in the more complicated methods (`fetch` and `install`). In `node.rs` and `yarn.rs`, all of the ones that are in the analogous methods are already single lines, so I left them alone.